frontend/src/pages/projects/screens/spawner/storage/AddExistingStorageFormField.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import { Alert, FormGroup } from '@patternfly/react-core';
+import { TypeaheadSelectOption } from '@patternfly/react-templates';
+import { StorageData } from '~/pages/projects/types';
+import { ProjectDetailsContext } from '~/pages/projects/ProjectDetailsContext';
+import { getDisplayNameFromK8sResource } from '~/concepts/k8s/utils';
+import TypeaheadSelect from '~/components/TypeaheadSelect';
+import useProjectPvcs from '~/pages/projects/screens/detail/storage/useProjectPvcs';
+
+type AddExistingStorageFormFieldProps = {
+  data?: StorageData['creating'];
+  setData: (data: StorageData['creating']) => void;
+  selectDirection?: 'up' | 'down';
+  menuAppendTo?: HTMLElement;
+};
+
+const AddExistingStorageFormField: React.FC<AddExistingStorageFormFieldProps> = ({
+  data,
+  setData,
+  selectDirection,
+  menuAppendTo,
+}) => {
+  const { currentProject } = React.useContext(ProjectDetailsContext);
+  const [storages, loaded, loadError] = useProjectPvcs(currentProject.metadata.name);
+
+  const selectDescription = (size: string, description?: string) => (
+    <div>
+      <div>Size: {size}</div>
+      {description && <div>Description: {description}</div>}
+    </div>
+  );
+
+  const selectOptions: TypeaheadSelectOption[] = React.useMemo(
+    () =>
+      loaded
+        ? storages.map((pvc) => ({
+            value: pvc.metadata.name,
+            content: getDisplayNameFromK8sResource(pvc),
+            description: selectDescription(
+              pvc.spec.resources.requests.storage,
+              pvc.metadata.annotations?.['openshift.io/description'],
+            ),
+          }))
+        : [],
+    [loaded, storages],
+  );
+
+  if (loadError) {
+    return (
+      <Alert title="Error loading pvcs" variant="danger">
+        {loadError.message}
+      </Alert>
+    );
+  }
+
+  let placeholderText: string;
+
+  if (!loaded) {
+    placeholderText = 'Loading storages...';
```

Comment:
I'm not sure "Loading storages" is what we want here. Maybe just 'Loading storage'. I would check with @xianli123 or @kaedward.